src/MimeToken.sol

```diff
@@ -10,10 +11,12 @@ error AlreadyClaimed();
 error InvalidProof();
 error NonTransferable();
 
-contract MimeToken is Ownable, IMimeToken {
+contract MimeToken is Initializable, OwnableUpgradeable, IMimeToken {
     uint256 private _currentRound;
     string private _name;
     string private _symbol;
+    uint256 private _timestamp;
```

Comment:
Maybe we could rename this to `_initialTimestamp` or `_startTimestamp`  to improve legibility. Wdyt?